graphql_compiler/macros/macro_edge/helpers.py

```diff
@@ -53,6 +60,143 @@ def get_directives_for_ast(ast):
     return result
 
 
+def get_all_tag_names(ast):
+    """Return a set of strings containing tag names that appear in the query.
+
+    Args:
+        ast: GraphQL query AST object
+
+    Returns:
+        set of strings containing tag names that appear in the query
+    """
+    return {
+        # Schema validation has ensured this exists
+        directive.arguments[0].value.value
+        for ast, directive in _yield_ast_nodes_with_directives(ast)
+        if directive.name.value == TagDirective.name
+    }
+
+
+def _replace_tag_names_in_tag_directive(name_change_map, tag_directive):
+    """Return the new directive, and whether it is different from the old."""
+    # Schema validation has ensured this exists
+    current_name = tag_directive.arguments[0].value.value
+    new_name = name_change_map[current_name]
+    renamed_tag_directive = copy(tag_directive)
+    renamed_tag_directive.arguments = [Argument(Name('tag_name'), StringValue(new_name))]
+    return renamed_tag_directive, new_name != current_name
+
+
+def _replace_tag_names_in_filter_directive(name_change_map, filter_directive, ast):
+    """Return the new directive, and whether it is different from the old."""
+    made_changes = False
+    filter_with_renamed_args = copy(filter_directive)
+    filter_with_renamed_args.arguments = []
+    for argument in filter_directive.arguments:
+        if argument.name.value == 'op_name':
+            filter_with_renamed_args.arguments.append(argument)
+        elif argument.name.value == 'value':
+            new_value_list = []
+            for value in argument.value.values:
+                if is_tag_argument(value.value):
+                    current_name = get_directive_argument_name(value.value)
+                    new_name = name_change_map[current_name]
+                    if new_name != current_name:
+                        made_changes = True
+                    new_value_list.append(StringValue('%' + new_name))
+                elif is_variable_argument(value.value):
+                    new_value_list.append(value)
+                else:
+                    raise AssertionError(u'Value was neither tag nor filter: {} {}'
+                                         .format(filter_directive, value.value))
+            filter_with_renamed_args.arguments.append(
+                Argument(Name('value'), value=ListValue(new_value_list)))
+        else:
+            raise AssertionError(u'Unknown argument name {} in filter: {}'
+                                 .format(argument.name.value, ast))
+    return filter_with_renamed_args, made_changes
+
+
+def _replace_tag_names_at_current_node(name_change_map, ast):
+    """Return a new ast with tag names replaced according to the name_change_map."""
```

Comment:
This docstring is very misleading. Having read this, I would expect the function's signature to be:
```
T = TypeVar('T', Field, OperationDefinition, InlineFragment)

def _replace_tag_names_at_current_node(name_change_map: Dict[str, str], ast: T) -> T:
    ...
```

However, the actual signature's return type is `-> Tuple[List[Directive], bool]`.

Please write out a full docstring, including `Args/Returns`.

Reply:
While we're at it, the function's name is also misleading in the same way.